client.go

```diff
@@ -0,0 +1,69 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"github.com/ONSdigital/log.go/log"
+	"time"
+
+	"github.com/ONSdigital/dp-sessions-api/session"
+	goredis "github.com/go-redis/redis"
+)
+
+var ctx = context.Background()
+
+// Client - structure for the redis client
+type Client struct {
+	client *goredis.Client
+	ttl    time.Duration
+}
+
+// Config - config options for the redis client
+type Config struct {
+	Addr     string
+	Password string
+	Database int
+	TTL      time.Duration
+}
+
+// NewClient - returns new redis client with provided config options
+func NewClient(c Config) (*Client, error) {
+	if c.Addr == "" {
+		return nil, errors.New("address is missing")
+	}
+
+	if c.Password == "" {
+		return nil, errors.New("password is missing")
+	}
+
+	return &Client{
+		client: goredis.NewClient(&goredis.Options{
```

Comment:
At the moment this client is tightly coupled to the `goredis` package. Consider defining your own interface type and taking it an instance as a parameter to  `NewClient()`.

This decouples this from the goredis package and allows you to more easily change it in future if needed and also allows you pass in a mock for testing.